tests/rptest/tests/connection_limits_test.py

```diff
@@ -46,6 +47,16 @@ def test_exceed_broker_limit(self):
         for c in consumers:
             c.start()
 
+        def connection_count():
+            return self.redpanda.metric_sum(ACTIVE_METRIC)
+
+        # wait until the connection count reaches the expected number before starting the
+        # producer, since otherwise the consumers and the producer race and the producer
+        # may win in which case it would be one of the consumers that fail to connect
+        self.redpanda.wait_until(
+            lambda: connection_count() == 6, 60, 1,
```

Comment:
any chance this becomes flaky? for example a once-in-a-while connection gets dropped and re-opened by franz-go/rpk but the metric doesn't update so quickly?

Reply:
I did consider the possible flakiness here.

For one thing, this just wouldn't work on a system where there are any unaccounted-for connections, e.g., in a cloud test with other stuff connected (say, kminion) this condition might simply fail (connections may never hit 6, they may be above even from the start, or jump from 1 to 4 to 7 or something like that. However, this test is already written in a way that expects a clean system since it counts connections "exactly".

The metric itself updates instantly, based on RP's view of the connections, so if a connection was silently dropped then it could continue to reflect a non-existent connection for a while. This doesn't seem _that_ likely as we are making fresh connections and waiting for the number to hit the expected count which generally happens almost instantly.

However, I think it would reduce the chance of future flakiness if I made this >= 6, rather than == 6, at the cost of not being informed about unexpected changes in the connection behavior. I think that's probably closer to the original intent of this test. I'll make that change unless anyone disagrees.

Reply:
Changed to `>= 6` in https://github.com/redpanda-data/redpanda/commit/cb22f0a216adfc2f7add23b5fcc691c6e431d6c7